core/loadpoint.go

```diff
@@ -718,18 +718,40 @@ func (lp *Loadpoint) syncCharger() error {
 			} else if !errors.Is(err, api.ErrNotAvailable) {
 				return fmt.Errorf("charger get max current: %w", err)
 			}
+		} else {
+			// use measured phase currents as fallback, validate if current too high by at least 1A
+			if current := lp.GetMaxPhaseCurrent(); current > lp.chargeCurrent+1.0 {
+				if shouldBeConsistent {
+					lp.log.WARN.Printf("charger logic error: current mismatch (got %.3gA measured, expected %.3gA)", current, lp.chargeCurrent)
+				}
+				lp.chargeCurrent = current
+				lp.bus.Publish(evChargeCurrent, lp.chargeCurrent)
+			}
 		}
 
 		// sync phases
 		phases := lp.GetPhases()
-		if ps, ok := lp.charger.(api.PhaseGetter); ok && enabled && shouldBeConsistent && phases > 0 {
-			if chargerPhases, err := ps.GetPhases(); err == nil {
-				if chargerPhases != phases {
-					lp.log.WARN.Printf("charger logic error: phases mismatch (got %d, expected %d)", chargerPhases, phases)
-					lp.setPhases(chargerPhases)
+		if shouldBeConsistent && phases > 0 {
+			// fallback active phases from measured phases
+			chargerPhases := lp.measuredPhases
+			if chargerPhases == 2 {
+				chargerPhases = 3
+			}
+
+			if ps, ok := lp.charger.(api.PhaseGetter); ok {
+				if cp, err := ps.GetPhases(); err == nil {
+					chargerPhases = cp
+				} else {
+					if errors.Is(err, api.ErrNotAvailable) {
+						return nil
+					}
+					return fmt.Errorf("charger get phases: %w", err)
 				}
-			} else if !errors.Is(err, api.ErrNotAvailable) {
-				return fmt.Errorf("charger get phases: %w", err)
+			}
+
+			if chargerPhases > 0 && chargerPhases != phases {
```

Comment:
What state of the charger would chargerPases = 0 indicate? Not connected?
I assume this is OK, but I just would like to understand why we only report the mismatch if chargerPhases > 0.

Reply:
It just indicates- in case we're relying on the fallback- that no value has been determined, i.e. because currents are not available (yet).

Reply:
PS.: good question :)